apps/src/code-studio/assets/SelectStartAnimations.jsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import {
+  getManifest,
+  generateLevelAnimationsManifest
+} from '@cdo/apps/assetManagement/animationLibraryApi';
+import AnimationPickerBody from '@cdo/apps/p5lab/AnimationPicker/AnimationPickerBody.jsx';
+import {createUuid} from '@cdo/apps/utils';
+import color from '@cdo/apps/util/color';
+import PropTypes from 'prop-types';
+import {PICKER_TYPE} from '@cdo/apps/p5lab/AnimationPicker/AnimationPicker';
+
+const THUMBNAIL_SIZE = 50;
+const THUMBNAIL_BORDER_WIDTH = 1;
+
+export default class SelectStartAnimations extends React.Component {
+  static propTypes = {
+    useAllSprites: PropTypes.bool
+  };
+
+  state = {
+    levelAnimationsManifest: {},
+    libraryManifest: {},
+    orderedKeys: [],
+    propsByKey: {}
+  };
+
+  componentDidMount() {
+    getManifest('spritelab')
+      .then(sprites => {
+        this.setState({libraryManifest: sprites});
+      })
+      .catch(err => {
+        console.log(err);
+      });
+
+    if (this.props.useAllSprites) {
+      generateLevelAnimationsManifest()
+        .then(manifest => {
+          this.setState({levelAnimationsManifest: JSON.parse(manifest)});
+        })
+        .catch(err => {
+          console.log(err);
+        });
+    }
+  }
+
+  addAnimationToList = animation => {
+    const key = createUuid();
+
+    let updatedOrderedKeys = [key].concat(this.state.orderedKeys);
+    this.setState({orderedKeys: updatedOrderedKeys});
+
+    let propsByKey = {...this.state.propsByKey};
+    propsByKey[key] = animation;
+    this.setState({propsByKey: propsByKey});
+  };
+
+  removeAnimationFromList = key => {
+    let updatedOrderedKeys = [...this.state.orderedKeys];
+    let indexToRemove = updatedOrderedKeys.indexOf(key);
+    updatedOrderedKeys.splice(indexToRemove, 1);
+    this.setState({orderedKeys: updatedOrderedKeys});
+
+    let propsByKey = {...this.state.propsByKey};
+    delete propsByKey[key];
+    this.setState({propsByKey: propsByKey});
+  };
+
+  displaySelectedSprites = () => {
+    const {propsByKey, orderedKeys} = this.state;
+    return orderedKeys.map(key => {
+      return (
+        <img
+          key={key}
+          src={propsByKey[key].sourceUrl}
+          alt={propsByKey[key].name}
+          style={styles.thumbnail}
+          role="button"
+          onClick={() => this.removeAnimationFromList(key)}
+        />
+      );
+    });
+  };
+
+  render() {
+    const {orderedKeys, propsByKey} = this.state;
+    return (
+      <div>
```

Comment:
now that we're on react 16, we can use a [fragment](https://reactjs.org/docs/fragments.html) here!